.github/workflows/test-python.yaml

```diff
@@ -5,25 +5,52 @@ on:
   push:
     branches: [main]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   test:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.9', '3.11']
+        dependency-group:
+          - {name: dev,  make-target: test-python-dev}
+          - {name: prod, make-target: test-python}
+
+    name: Test ${{ matrix.dependency-group.name }} (Python ${{ matrix.python-version }})
+
     steps:
       - uses: actions/checkout@v4
 
-      - name: Set up Python
+      - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v5
         with:
-          python-version: '3.9'
+          python-version: ${{ matrix.python-version }}
 
       - name: Verify
         run: make verify
 
       - name: Run Tests
         run: |
-          make test-python report=xml
+          make ${{ matrix.dependency-group.make-target }} report=xml
 
       - name: Upload coverage to Coveralls
         uses: coverallsapp/github-action@v2
         with:
-          github-token: ${{ secrets.GITHUB_TOKEN }} # Pass the GITHUB_TOKEN
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          flag-name: ${{ matrix.dependency-group.name }}-py${{ matrix.python-version }}
+          parallel: true
+
+  coverage:
```

Comment:
Don't we run coverage as part of the previous job in Upload coverage to Coveralls step ?

Reply:
As I can see, it shows the report on the PRs: https://github.com/kubeflow/sdk/pull/56#issuecomment-3144804385